### PR TITLE
Reworking Bot API and optimizing for loops in admin code

### PR DESCRIPTION
**Pull request by Sindorman**
This PR goes along with https://github.com/Aurorastation/BOREALISbot2/pull/28

It:

- Reworks API to count staff and admins. Makes them as separate fields for bot to display

- Renames global list `admin` to `staff` since it includes anyone with perms.

- Optimizes all for loops that iterate over client lists, reducing byond in-built `istype()` check.

### CODE REVIEW

File: code/modules/world_api/commands/server_query.dm [hunk 1]
```diff
@@ -21,12 +21,11 @@
 
 	s["players"] = clients.len
 	s["admins"] = 0
+	s["staff"] = staff.len
 
-	for(var/client/C in clients)
-		if(C.holder)
-			if(C.holder.fakekey)
```
Comment (Arrow768) [inline code/modules/world_api/commands/server_query.dm]: If holder.fakekey is set that person should not be counted.

File: code/modules/world_api/commands/server_query.dm [hunk 2]
```diff
@@ -21,12 +21,11 @@
...
-				continue
-
+	for(var/S in staff)
+		var/client/C = S
+		if(C.holder.rights & (R_MOD|R_ADMIN))
```
Comment (Arrow768) [inline code/modules/world_api/commands/server_query.dm]: I am pretty sure that this binary expression is wrong as that would only count people who have R_MOD and R_ADMIN

(R_MOD|R_ADMIN) results in a binary value containing both R_MOD and R_ADMIN

It should be sufficient to check for R_MOD as admins also have R_MOD

Comment (Sindorman) [reply code/modules/world_api/commands/server_query.dm]: Well me and Skull were not sure if admins have R_MOD. Plus I took it as all other code uses R_MOD|R_ADMIN

Comment (Sindorman) [reply code/modules/world_api/commands/server_query.dm]: Alberyk does not want retired admins to be counted, and they get both R_MOD and R_ADMIN, but not R_BAN which both admins and mods have.

**Pull request merged**